### PR TITLE
Refactor a bunch of analyzer projects into Microsoft.CodeQuality.Anal…

### PR DESCRIPTION
**Pull request by mavasani**
…yzers

We have decided to change the grouping structure of the analyzer projects as follows:

1. **Microsoft.CodeQuality.Analyzers** (final name not decided): These will contain common analyzers to improve code quality. They will contain:
   1. ApiReview.Analyzers
   2. ApiDesignGuidelines.Analyzers
   3. Microsoft.QualityGuidelines.Analyzers
   4. Microsoft.Maintainability.Analyzers
   5. XmlDocumentationComment.Analyzers
2. **NETCore.Analyzers**: Analyzers specific to usage of APIs from specific netcore libraries. These will contain:
   1. Microsoft.Composition.Analyzers
   2. System.Collection.Immutable.Analyzers
   3. System.Resources.Analyzers
   4. System.Runtime.Analyzers
   5. System.Runtime.InteropService.Analyzers
   6. System.Threading.Tasks.Analyzers
   7. System.Security.Cryptography.Hashing.Algorithms.Analyzers
3. **Desktop.Analyzers**: Analyzers specific to usage of APIs from .NET framework libraries (will probably be renamed to NETFramework.Analyzers in future).
4. **Microsoft.CodeAnalysis.Analyzers**: These are analyzers for Microsoft.CodeAnalysis package
5. **Rolsyn.Diagnostic.Analyzers**: These are internal to Roslyn and non-shipping.

This change includes the following:
1. Implement the first step of above refactoring and create **Microsoft.CodeQuality.Analyzers** projects. Currently, they are just linking to the existing source files, but soon we will move the source files directly under this project.
2. Add capability to build the analyzers with Suggestion/Info diagnostics by default, instead of Warning diagnostics. This enables us to dogfood the analyzer VSIXes without producing breaking diagnostics/noise in the IDE error list.
3. Add capability to build the analyzer assemblies using the internal IOperation APIs that by-pass the IOperation feature flag to enable wider dogfooding.

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @dotnet/analyzer-ioperation @jinujoseph for review

File: src/System.Threading.Tasks.Analyzers/Setup/source.extension.vsixmanifest [hunk 1]
```diff
@@ -13,9 +13,9 @@
     <Dependency Id="Microsoft.Framework.NDP" DisplayName="Microsoft .NET Framework" d:Source="Manual" Version="[4.5,)" />
   </Dependencies>
   <Assets>
-    <Asset Type="Microsoft.VisualStudio.Analyzer" Path="System.Threading.Tasks.Analyzers.dll" />
-    <Asset Type="Microsoft.VisualStudio.Analyzer" Path="System.Threading.Tasks.CSharp.Analyzers.dll" />
-    <Asset Type="Microsoft.VisualStudio.Analyzer" Path="System.Threading.Tasks.VisualBasic.Analyzers.dll" />
+    <Asset Type="Microsoft.VisualStudio.Analyzer" Path="System.Threading.Tasks.Analyzers.dll" IsOptionBased="true" />
```
Comment (jinujoseph) [inline src/System.Threading.Tasks.Analyzers/Setup/source.extension.vsixmanifest]: Nit: We should add a comment about the purpose of IOptionBased.

Comment (mavasani) [inline src/System.Threading.Tasks.Analyzers/Setup/source.extension.vsixmanifest]: I think we probably need to first finalize the design here? Can we discuss this at the next CodeAnalysis design meeting?

**Comment by jinujoseph** [comment]
LGTM

File: src/System.Threading.Tasks.Analyzers/Setup/source.extension.vsixmanifest (hunk 1 above)
Comment (jinujoseph) [inline src/System.Threading.Tasks.Analyzers/Setup/source.extension.vsixmanifest]: Sure

**Pull request merged**